### PR TITLE
TSK-1506: Fixed bug for Cleanup Jobs with more than 32767 tasks

### PR DESCRIPTION
**Pull request by tge20**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [x] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Comment by tge20** [comment]
https://sonarcloud.io/dashboard?branch=TSK-1506&id=tge20_taskana

File: common/taskana-common/src/test/java/pro/taskana/common/api/ListUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.junit.jupiter.api.Test;
+
+import pro.taskana.common.internal.util.ListUtil;
+
+/** Test for the ListUtil class. */
+class ListUtilTest {
+
+  @Test
+  void should_SplitListIntoChunks_When_CallingPartitionBasedOnSize() {
+    List<Integer> collect = IntStream.rangeClosed(1, 1000).boxed().collect(Collectors.toList());
```
Comment (mustaphazorgati) [inline common/taskana-common/src/test/java/pro/taskana/common/api/ListUtilTest.java:18]: please rename this variable to something useful. Like `ListWith1000Entries`

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQueryImpl.java [hunk 2]
```diff
@@ -161,7 +161,7 @@
   private boolean addAttachmentClassificationNameToSelectClauseForOrdering = false;
   private boolean addWorkbasketNameToSelectClauseForOrdering = false;
 
-  TaskQueryImpl(InternalTaskanaEngine taskanaEngine) {
+  public TaskQueryImpl(InternalTaskanaEngine taskanaEngine) {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQueryImpl.java:164]: This is not necessary anymore, right?

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java [hunk 3]
```diff
@@ -992,18 +994,26 @@ void removeNonExistingTasksFromTaskIdList(
...
+  private List<TaskSummaryImpl> augmentTaskSublist(List<TaskSummaryImpl> taskSummaries) {
     List<String> taskIds =
         taskSummaries.stream().map(TaskSummaryImpl::getId).distinct().collect(Collectors.toList());
 
     if (taskIds.isEmpty()) {
       taskIds = null;
     }
+
     LOGGER.debug(
         "augmentTaskSummariesByContainedSummaries() about to query for attachmentSummaries ");
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:1018]: debug statement and method name does not match.

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java [hunk 4]
```diff
@@ -992,18 +994,26 @@ void removeNonExistingTasksFromTaskIdList(
...
 
-    List<TaskSummary> result = new ArrayList<>();
-    if (taskSummaries == null || taskSummaries.isEmpty()) {
-      return result;
-    }
+    LOGGER.debug("exit from to augmentTaskSummariesByContainedSummaries()");
+    return result;
+  }
 
+  private List<TaskSummaryImpl> augmentTaskSublist(List<TaskSummaryImpl> taskSummaries) {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:1009]: I don't like this name. Maybe we can call this `augmentTaskSummariesByContainedSummariesWithoutPartitioning` and the parent method `augmentTaskSummariesByContainedSummariesWithPartitioning`

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQueryImpl.java (hunk 2 above)
Comment (tge20) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQueryImpl.java:164]: Yes, that's right.

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java (hunk 4 above)
Comment (tge20) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:1009]: What about "augmentTaskSummariesByContainedSummaries" and "augmentTaskSummariesByContainedSummariesForSublist"? Or do you prefer your suggestions? I'm not a huge fan of complex method names...

File: common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  /**
+   * Splits a list with objects of type T into chunks of a certain size.
+   *
+   * @param <T> type of elements inside list
+   * @param inputList list to be divided
+   * @param size maximal number of elements inside chunk
+   * @return list containing the chunks
+   */
+  public static <T> Collection<List<T>> partitionBasedOnSize(Collection<T> inputList, int size) {
```
Comment (mustaphazorgati) [inline common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java:22]: `Collection<T> inputList` seems a little confusing ;)

Comment (mustaphazorgati) [reply common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java:22]: I'd name it `inputCollection`. But no need to change this here (you can do that in the next PR)

File: common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package pro.taskana.common.internal.util;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+public class ListUtil {
```
Comment (mustaphazorgati) [inline common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java:8]: Didn't we agree on `CollectionUtil`?

**Review by mustaphazorgati — APPROVED** [review]

File: common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java (hunk 5 above)
Comment (tge20) [reply common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java:22]: Right, I do it on the next PR

File: common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java (hunk 6 above)
Comment (tge20) [reply common/taskana-common/src/main/java/pro/taskana/common/internal/util/ListUtil.java:8]: I do it on the next PR.

**Pull request merged**